discord/message.py

```diff
@@ -208,6 +208,33 @@ async def to_file(self, *, use_cached=False, spoiler=False):
         data = await self.read(use_cached=use_cached)
         return File(io.BytesIO(data), filename=self.filename, spoiler=spoiler)
 
+class MessageReference:
+    """Represents a reference to a :class:`Message`.
+
+    .. versionadded:: 1.5
+
+    Attributes
+    -----------
+    message_id: Optional[:class:`int`]
+        The id of the message referenced.
+    channel_id: :class:`int`
+        The channel id of the message referenced.
+    guild_id: Optional[:class:`int`]
+        The guild id of the message referenced.
+    cached_message: Optional[:class:`Message`]
+        The cached message, if found in the internal message cache.
+    """
+    __slots__ = ('message_id', 'channel_id', 'guild_id', 'cached_message')
+
+    def __init__(self, **kwargs):
+        self.message_id = int(kwargs.pop('message_id', 0)) or None
+        self.channel_id = int(kwargs.pop('channel_id'))
+        self.guild_id = int(kwargs.pop('guild_id', 0)) or None
+        self.cached_message = None
```

Comment:
I'd prefer if this was a dynamic lookup property rather than a strong reference.

Reply:
I'm unclear on the reasoning behind dynamic lookup on properties aside from the cached message.
Resolved.